### PR TITLE
Add overflow wrapping support

### LINKED ISSUE #54: Add multiline/overflow wrapping support
I've put together a POC for multiline/overflow wrapping support for `react-logviewer`: https://github.com/HunterCFigueroa/react-logviewer/tree/huntercfigueroa/multiline-wrapping

It's working quite nicely from what i can tell and I wanted to gauge your interest.

Key Changes:
- Replaces `react-window` with `virtua` for it's built-in auto sizing, this does most of the work
- New styles for lines, content, and parts
- style changes gated by `wrapLines` prop

### PR DESCRIPTION
**Pull request by HunterCFigueroa**
Fix #54

Replaces react-window with virtua. Virtua has a some really nice autosizing built in which opens up the opportunity for overflow wrapping.

Not sure if the passed down wrapLines prop is the ideal approach but it gets the job done.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -44,7 +44,7 @@
         "mitt": "^3.0.1",
         "react-string-replace": "^1.1.1",
         "react-virtualized-auto-sizer": "^1.0.24",
```
Comment (melloware) [inline package.json:46]: Can we also remove `react-virtualized-auto-sizer`?

File: src/components/LazyLog/index.tsx [hunk 2]
```diff
@@ -1180,65 +1194,19 @@ export default class LazyLog extends Component<LazyLogProps, LazyLogState> {
...
-                {/*
-                 // @ts-ignore */}
-                <AutoSizer
-                    disableHeight={this.props.height !== "auto"}
-                    disableWidth={this.props.width !== "auto"}
+                <VList
+                    className={`react-lazylog ${styles.lazyLog} ${
+                        this.props.wrapLines ? styles.wrap : ""
+                    }`}
+                    style={{ height: 800 }}
```
Comment (melloware) [inline src/components/LazyLog/index.tsx:1201]: Hardcoded 800? Instead of calculateListHeight?

File: src/components/LazyLog/index.tsx [hunk 3]
```diff
@@ -1180,65 +1194,19 @@ export default class LazyLog extends Component<LazyLogProps, LazyLogState> {
...
-                                }
-                                itemSize={this.getItemSize}
-                                initialScrollOffset={this.state.scrollToIndex}
-                                itemCount={
-                                    rowCount === 0
-                                        ? rowCount
-                                        : rowCount +
-                                          (this.props.extraLines || 0)
-                                }
-                                onScroll={(options) => {
```
Comment (melloware) [inline src/components/LazyLog/index.tsx:1211]: It looks like you removed onScroll and the ability for users to attach to onScroll event?  Also what about ScrollFollowing?

**Review by melloware — CHANGES_REQUESTED** [review]

File: src/components/LazyLog/index.tsx (hunk 2 above)
Comment (HunterCFigueroa) [reply src/components/LazyLog/index.tsx:1201]: Yeah, that was just for getting the POC up and running. Now that I know there's interest Ill spend some time cleaning it up.

**Comment by melloware** [comment]
Also to answer your question about `wrapLines` bubbling down that is the style of the component when I took it over so i like that you kept that pattern.    Overall i feel like your changes will be a big improvement as Virtua looks pretty impressive.

**Comment by melloware** [comment]
Also it looks like Virtua has some nice features to replace current features of React Window:

ScrollTo: https://inokawa.github.io/virtua/?path=/story/basics-vlist--scroll-to
Scroll Restoration: https://inokawa.github.io/virtua/?path=/story/basics-vlist--scroll-restoration
Infinite Scrolling: https://inokawa.github.io/virtua/?path=/story/basics-vlist--infinite-scrolling

**Comment by HunterCFigueroa** [comment]
Alright i think i got it all working, but you would know better than me lol

**Comment by melloware** [comment]
This is great let me do some testing!

**Comment by melloware** [comment]
In the highlighting and scrolling example the correct values are being highlihgted but not automatically scrolled to.

See:
image

compare with: https://melloware.github.io/react-logviewer/?path=/story/stories-lazylog--highlight-and-scrolling

**Review by melloware — CHANGES_REQUESTED** [review]
See comments

**Comment by HunterCFigueroa** [comment]
Nice catch 🐟

**Comment by melloware** [comment]
also any chance you could add a new storybook showing the line wrapping on?

**Comment by melloware** [comment]
This is looking pretty good so far i think this will be a good long term change to the component!

**Comment by melloware** [comment]
ok nice!  Is this safe for me to re-review?

File: src/components/LazyLog/index.module.css [hunk 4]
```diff
@@ -15,6 +15,10 @@
     color: #d6d6d6;
 }
 
+.wrap {
```
Comment (melloware) [inline src/components/LazyLog/index.module.css:18]:
```suggestion
.wrapLine {
```

File: src/components/LazyLog/index.module.css.d.ts [hunk 5]
```diff
@@ -3,6 +3,7 @@ declare namespace IndexModuleCssNamespace {
         lazyLog: string;
         searchMatch: string;
         searchMatchHighlighted: string;
+        wrap: string;
```
Comment (melloware) [inline src/components/LazyLog/index.module.css.d.ts:6]:
```suggestion
        wrapLine: string;
```

File: src/components/LazyLog/index.tsx [hunk 6]
```diff
@@ -347,6 +349,7 @@ export default class LazyLog extends Component<LazyLogProps, LazyLogState> {
         },
         caseInsensitive: false,
         enableGutters: false,
+        wrapLines: true,
```
Comment (melloware) [inline src/components/LazyLog/index.tsx:352]: Move this down after the last `enable` so its not mixed in the middle and more alpha order?

**Review by melloware — CHANGES_REQUESTED** [review]

File: src/components/LazyLog/index.tsx [hunk 7]
```diff
@@ -376,6 +378,7 @@ export default class LazyLog extends Component<LazyLogProps, LazyLogState> {
         eventsource: false,
         eventsourceOptions: {},
         width: "auto",
+        wrapLines: true,
```
Comment (HunterCFigueroa) [inline src/components/LazyLog/index.tsx:381]: Would you like it disabled by default?

**Comment by HunterCFigueroa** [comment]
Ready for re-review @melloware

File: src/components/LazyLog/index.tsx (hunk 7 above)
Comment (melloware) [reply src/components/LazyLog/index.tsx]: Yes disabled by default would be ideal.

**Review by melloware — APPROVED** [review]

**Comment by melloware** [comment]
I think this is good enough to merge.  Did you want to look at https://github.com/melloware/react-logviewer/issues/8 next since i think now it can be supported easily with Virtua?

**Comment by HunterCFigueroa** [comment]
Sure, I can take a look

**Comment by melloware** [comment]
also can you check the HTML links example?  I am not sure why the spacing is so wide and whether it was before?

image

**Comment by HunterCFigueroa** [comment]
Yeah it wasn't like that before, looking now

**Pull request merged**